src/cpp/lpnamer/problem_modifier/include/antares-xpansion/lpnamer/problem_modifier/RenameUtils.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <mutex>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
+
+class RenameUtils
+{
+public:
+    bool try_replace_first_token(const std::string& name,
+                                 std::string_view prefix,
+                                 unsigned int week,
+                                 long long factor,
+                                 bool ignore_value,
+                                 std::string& out) const;
+
+    std::string replace_hour_in_name(const std::string& name, unsigned int week) const;
+    std::pair<std::vector<std::string>&, std::vector<std::string>&> rename_week_names(
+      unsigned int week,
+      const std::vector<std::string>& variables,
+      const std::vector<std::string>& contraintes) const;
+
+    void rename_week_names(
+      unsigned int week,
+      const std::vector<std::string>& names,
+      std::unordered_map<int, std::vector<std::string>>& container_names) const;
```

Comment:
Make this method `private` to avoid confusion with the other overload
```suggestion
    static void rename_week_names(
      unsigned int week,
      const std::vector<std::string>& names,
      std::unordered_map<int, std::vector<std::string>>& container_names) const;
```